free-2.md

```diff
@@ -7,37 +7,75 @@ good way to meet their needs.
-_Describe the project's domain in five words._
+“Guitar chord”-based piano accompaniment generator.
 
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
+_How might the user interact with the language?_
+
+I’m envisioning for the user to look up chord tabs such as the one below:
+
+http://www.heartwoodguitar.com/chords/swift-taylor-you-belong-with-me/
+
+Given that the duration of each chord changes (for example, the D Major chord is played during the lyrics “You’re on the phone with your girlfriend, she’s upset” and then transitions to the A Major chord), I want the user to use an excel spreadsheet, write down the time signature and tempo of the song, and treat each excel cell as one beat. Then the user would input the name of the chord, each time it changes, onto the excel spreadsheet, taking into account the fact that each excel spreadsheet represents one beat. The user will then save the excel file as in some format that my language will be able to parse, and my language will communicate with a software such as MuseScore to produce the sheet music dictation of the chords and their respective lengths. I also want to be able to somehow incorporate lyrics so that onto the sheet music so that the user can sing along with the chords that are now dictated on a musical staff. I think this is the right way to interact with the problem domain, because I want to make sure that using this program requires as little musical theory knowledge as possible.
+
+_What does programming look like? Why is this the right way to interact with the problem domain?_ 
+
+Programming involves inputting chord names into an excel spreadsheet, where each cell represents one beat of the song. This is the right way to interact with the problem domain because people can visually represent the duration of the chords and people can simply copy and paste the chord names they want, either from existing guitar chords sheets or chords that they themselves want to incorporate into the song. Using this program won't require much music theory knowledge!
 
 
 ### Operation (semantics)
-_What might happen when a program runs? How does a program interact with the
-user? What kinds of errors might occur, and how might they be communicated to
-the user?_
+_What might happen when a program runs?_
+
+The program will ask the user to upload a xsl file, and then the program will parse the file, create the sheet music score in a program like MuseScore, inform the user that it is loading, and then allow the user to download the MuseScore file when the program is done.
+
+_How does a program interact with the user?_
 
 
+
+_What kinds of errors might occur, and how might they be communicated to the user?_
+
+One error I anticipate is if the uploaded file is the wrong file, and the program would show the error message “please make sure that you are uploading a xsl file”.
+
```

Comment:
What would happen if the user entered in an invalid chord?  Would you want the program to compile and display as much information as possible, or indicate that there was an error without compiling?  I also wonder if you could get any alignment errors.  Ideally the lyrics would match up with the chords, but what if a user typed 100 words into a single cell?